src/components/tabBar/TabBarItem.js

```diff
@@ -37,36 +41,79 @@ export default class TabBarItem extends BaseComponent {
      * callback for when pressing a tab
      */
     onPress: PropTypes.func,
+    /**
+     * whether the tab will have a divider on its right
+     */
+    divider: PropTypes.bool,
+    /**
+     * A fixed width for the item
+     */
+    width: PropTypes.number,
+    /**
+     * A callback to invoke for onLayout event
+     */
+    onLayout: PropTypes.func,
+  };
+
+  static defaultProps = {
+    maxLines: 1,
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      fontStyle: this.getFontStyle(props),
+    };
+  }
+
+  // HACK: for indicator width in TabBar
+  getFontStyle(props) {
+    const userProp = props.selectedLabelStyle ? props.selectedLabelStyle.fontWeight : undefined;
```

Comment:
this could break if the style has passed as number (when passing a StyleSheet object the style translate to a number). 
you can use flatten for this https://facebook.github.io/react-native/docs/stylesheet.html#flatten